--- ogrsqlitevirtualogr.cpp	2019-09-06 09:14:10.000000000 +0200
+++ ../../../../build-arm64/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp	2019-09-06 15:37:25.000000000 +0200
@@ -71,7 +71,7 @@
 
 void OGR2SQLITE_Register()
 {
-    // sqlite3_auto_extension ((void (*)(void)) OGR2SQLITE_static_register);
+    sqlite3_auto_extension ((void (*)(void)) OGR2SQLITE_static_register);
 }
 
 #define VIRTUAL_OGR_DYNAMIC_EXTENSION_ENABLED
@@ -2587,7 +2587,7 @@
 {
     const sqlite3_api_routines * pApi = (const sqlite3_api_routines * )_pApi;
 #ifndef WIN32
-    if( pApi->create_module == nullptr )
+    if( ( pApi == nullptr ) || ( pApi->create_module == nullptr ) )
     {
         pApi = &OGRSQLITE_static_routines;
     }
