--- a/configure
+++ b/configure
@@ -38843,7 +38843,7 @@ $as_echo "$as_me: WARNING: GEOS was foun
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c -lgeos"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
@@ -38858,7 +38858,7 @@ if ${ac_cv_lib_geos_c_GEOSversion+:} fal
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lgeos_c
+LIBS="-lgeos_c -lgeos
        $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
