--- a/src/CMakeLists.txt	2022-08-04 14:59:47.000000000 +0200
+++ b/src/CMakeLists.txt	2022-08-04 15:00:53.000000000 +0200
@@ -44,7 +44,7 @@ SET( SOURCES
 	qca_securelayer.cpp
 	qca_securemessage.cpp
 	support/qpipe.cpp
-	support/console.cpp
+	# support/console.cpp
 	support/synchronizer.cpp
 	support/dirwatch.cpp
 	support/syncthread.cpp
@@ -52,8 +52,8 @@ SET( SOURCES
 
 IF (WIN32)
   SET( SOURCES ${SOURCES} qca_systemstore_win.cpp )
-elseif(APPLE)
-  set( SOURCES ${SOURCES} qca_systemstore_mac.cpp)
+# elseif(APPLE)
+#  set( SOURCES ${SOURCES} qca_systemstore_mac.cpp)
 else()
   SET( SOURCES ${SOURCES} qca_systemstore_flatfile.cpp )
 endif()
