--- a/configure	2022-05-17 14:24:24.000000000 +0200
+++ b/configure	2022-05-17 14:25:11.000000000 +0200
@@ -18266,8 +18266,8 @@ ac_res=$ac_cv_search_GEOSCoveredBy
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-else
-  as_fn_error $? "could not find libgeos_c (or obsolete 'libgeos_c' < v.3.3.0 found) - you may need to specify the directory of a geos-config file using --with-geosconfig" "$LINENO" 5
+#else
+#  as_fn_error $? "could not find libgeos_c (or obsolete 'libgeos_c' < v.3.3.0 found) - you may need to specify the directory of a geos-config file using --with-geosconfig" "$LINENO" 5
 fi
 
   LIBS="$LIBS_SAVE"
@@ -18353,8 +18353,8 @@ ac_res=$ac_cv_search_GEOSDelaunayTriangu
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-else
-  as_fn_error $? "'libgeos_c' (>= v.3.4.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geosadvanced parameter." "$LINENO" 5
+#else
+#  as_fn_error $? "'libgeos_c' (>= v.3.4.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geosadvanced parameter." "$LINENO" 5
 fi
 
 	  $as_echo "#define GEOS_ADVANCED 1" >>confdefs.h
@@ -18426,8 +18426,8 @@ ac_res=$ac_cv_search_GEOSContext_setErro
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-else
-  as_fn_error $? "'libgeos_c' (>= v.3.5.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geosreentrant parameter." "$LINENO" 5
+#else
+#  as_fn_error $? "'libgeos_c' (>= v.3.5.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geosreentrant parameter." "$LINENO" 5
 fi
 
 	  $as_echo "#define GEOS_REENTRANT 1" >>confdefs.h
@@ -18515,8 +18515,8 @@ ac_res=$ac_cv_search_GEOSFrechetDistance
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-else
-  as_fn_error $? "'libgeos_c' (>= v.3.7.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geos370 parameter." "$LINENO" 5
+#else
+#  as_fn_error $? "'libgeos_c' (>= v.3.7.0) is required but it doesn't seem to be installed on this system. You may need to try re-running configure with a --disable-geos370 parameter." "$LINENO" 5
 fi
 
 	  $as_echo "#define GEOS_370 1" >>confdefs.h