--- a/configure
+++ b/configure
@@ -12535,17 +12535,8 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu)
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  case $host_os in
-  # This must be Linux Android ELF.
-  linux-android*)
-    library_names_spec='$libname${shared_ext}'
-    soname_spec='${libname}${shared_ext}'
-    ;;
-  *)  
-    library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
-    soname_spec='${libname}${release}${shared_ext}$major'
-    ;;
-  esac
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
   finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
@@ -16381,17 +16372,8 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu)
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  case $host_os in
-  # This must be Linux Android ELF.
-  linux-android*)
-    library_names_spec='$libname${shared_ext}'
-    soname_spec='${libname}${shared_ext}'
-    ;;
-  *)  
-    library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
-    soname_spec='${libname}${release}${shared_ext}$major'
-    ;;
-  esac
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
   finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
@@ -17635,7 +17617,7 @@ $as_echo_n "checking for library contain
 if ${ac_cv_search_proj_normalize_for_visualization+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  ac_func_search_save_LIBS="$LIBS -ltiff -lwebp"
+  ac_func_search_save_LIBS=$LIBS
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17714,7 +17696,7 @@ for ac_lib in '' proj; do
     ac_res="none required"
   else
     ac_res=-l$ac_lib
-    LIBS="-l$ac_lib -lm -lsqlite3 -ltiff -lwebp $ac_func_search_save_LIBS"
+    LIBS="-l$ac_lib -lm -lpthread -lsqlite3 -ltiff -lwebp $ac_func_search_save_LIBS"
   fi
   if ac_fn_c_try_link "$LINENO"; then :
   ac_cv_search_pj_init_plus=$ac_res
@@ -18228,7 +18210,7 @@ done
 
   CPPFLAGS="$CPPFLAGS_SAVE"
   # Ensure we can link against libgeos_c
-  LIBS_SAVE="$LIBS -lgeos"
+  LIBS_SAVE="$LIBS -geos"
   LIBS="$GEOS_LDFLAGS"
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing GEOSCoveredBy" >&5
 $as_echo_n "checking for library containing GEOSCoveredBy... " >&6; }
